src/shared/main/java/com/couchbase/lite/FileLogger.java

```diff
@@ -113,19 +120,37 @@ public void setConfig(@Nullable LogFileConfiguration newConfig) {
             return;
         }
 
-        C4Log.writeToBinaryFile(
-            logDirPath,
-            logLevel.getValue(),
-            config.getMaxRotateCount(),
-            config.getMaxSize(),
-            config.usesPlaintext(),
-            CBLVersion.getVersionInfo());
+        config = newConfig.readOnlyCopy();
+
+        initLog();
     }
 
     @VisibleForTesting
     void reset() {
         config = null;
+        initializedPath = null;
         logLevel = LogLevel.NONE;
     }
+
+    private boolean initLog() {
+        final LogLevel level = logLevel;
+        final LogFileConfiguration cfg = config;
+
+        if ((cfg == null) || (level == LogLevel.NONE)) { return false; }
+
+        final String logDirPath = cfg.getDirectory();
+        if (logDirPath.equals(initializedPath)) { return false; }
```

Comment:
If setting the config before setting the level, I think this will cause the waring message to show up even though the config is not null?

Reply:
Not clear about this comment.  This method doesn't affect the warning message...

Reply:
I read the code wrong. Sorry. There is no issue here.